src/BKTConfig.ts

```diff
@@ -35,16 +44,28 @@ export interface BKTConfig extends RawBKTConfig {
   pollingInterval: number
   userAgent: string
   fetch: FetchLike
+  storageFactory: <T>(key: string) => BKTStorage<T>
+}
+
+const defaultUserAgent = () => {
+  if (typeof window === 'undefined') {
+    return `Bucketeer JavaScript SDK(${VERSION})`
+  } else {
+    return window.navigator.userAgent
+  }
 }
 
 export const defineBKTConfig = (config: RawBKTConfig): BKTConfig => {
+  const userAgent = defaultUserAgent()
+
   const result: BKTConfig = {
     eventsFlushInterval: MINIMUM_FLUSH_INTERVAL_MILLIS,
     eventsMaxQueueSize: DEFAULT_MAX_QUEUE_SIZE,
     pollingInterval: DEFAULT_POLLING_INTERVAL_MILLIS,
     storageKeyPrefix: '',
-    userAgent: window.navigator.userAgent,
-    fetch: window.fetch,
+    userAgent,
+    fetch,
+    storageFactory: createBKTStorage,
```

Comment:
By default the SDK checks if `localStorage` API exists and automatically change BKTStorage implementation(`localStorage` or in-memory)